File: algo_problems/q241-260/q247.py
class Solution:
    def findStrobogrammatic(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        return self.helper(n, n)

    def helper(self, n, m):
        if n == 0:
            return ['']
        if n == 1:
            return ['1', '8', '0']
        preList = self.helper(n - 2, m)
        result = []
        for s in preList:
            if n != m:
                result.append('0' + s + '0')
            result.append('1' + s + '1')
            result.append('8' + s + '8')
            result.append('6' + s + '9')
            result.append('9' + s + '6')
        return result


if __name__ == '__main__':
    print(Solution().findStrobogrammatic(2))
    print(Solution().findStrobogrammatic(3))

File: algo_problems/q241-260/q247.java
class Solution {
    public List<String> findStrobogrammatic(int n) {
        return this.helper(n, n);
    }

    private List<String> helper(int n, int m) {
    	if (n == 0) {
    		return new ArrayList<String>(Array.asList(""));
    	}
    	if (n == 1) {
    		return new ArrayList<String>(Array.asList("1", "8", "0"));
    	}

    	List<String> preList = this.helper(n - 2, m);
    	List<String> result = new ArrayList();
    	for (String s: preList) {
    		if(n != m)result.add("0" + s + "0");
    		result.add("1" + s + "1");
    		result.add("8" + s + "8");
    		result.add("6" + s + "9");
    		result.add("9" + s + "6");
    	}
    	return result;
    }
}